plugins/jenkins/komand_jenkins/actions/build_job/action.py

```diff
@@ -14,10 +17,29 @@ def __init__(self):
         )
 
     def run(self, params={}):
+        # START INPUT BINDING - DO NOT REMOVE - ANY INPUTS BELOW WILL UPDATE WITH YOUR PLUGIN SPEC AFTER REGENERATION
         name = params.get(Input.NAME)
-        parameters = params.get(Input.PARAMETERS)
+        parameters = params.get(Input.PARAMETERS, {}) or None
+        # END INPUT BINDING - DO NOT REMOVE
 
-        job_number = self.connection.server.build_job(name, parameters)
-        build_number = self.connection.server.get_job_info(name)["lastCompletedBuild"]["number"]
+        try:
+            # Build endpoint URL and cutoff parameters from that
+            endpoint = self.connection.server.build_job_url(name, parameters)
 
-        return {Output.JOB_NUMBER: job_number, Output.BUILD_NUMBER: build_number}
+            # Send request with parameters as 'data' instead of in URL as it's in package
+            response = self.connection.server.jenkins_request(
+                Request("POST", urljoin(endpoint, urlparse(endpoint).path), data=parameters)
+            )
+
+            # Retrieve the Job Number from Location header
+            location = response.headers.get("Location", "")
+
+            # Remove '/' characters if location ends with it
+            if location.endswith("/"):
+                location = location[:-1]
+
+            job_number = int(location.split("/")[-1])
+            build_number = self.connection.server.get_job_info(name).get("lastCompletedBuild", {}).get("number")
+            return {Output.JOB_NUMBER: job_number, Output.BUILD_NUMBER: build_number + 1}
```

Comment:
if the jenkins job runs really quickly could this `get_job_info` not return the job that was just triggered in the action?

Reply:
it returns current state and details related to specific job. I updated `lastCompletedBuild` to `lastBuild` to match 1:1 the same build number from UI as from plugin's response.I will push it 😄